### PR TITLE
debug CEF + add parsing beginning 2/18 samples v1

### PR DESCRIPTION
**Pull request by naegelnajim**
Debugging of CEF parsing initial format , 2 of 18 samples traited ECS fields v1 : beginning of the format

### CODE REVIEW

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+  set_data_loss_prevention_event:
+    actions:
+      - set:
+          event.kind: alert
+          event.category: ["email"]
+          event.type: ["info"]
+          source.ip: "{{parsed_event.message.src}}"
+          source.mac: "{{parsed_event.message.smac}}"
+          host.name: "{{parsed_event.message.shost}}"
+          user.name: "{{parsed_event.message.cs4Label}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:276]:
```suggestion
          user.name: "{{parsed_event.message.get('Incident_Source_(AD_Account)')}}"
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+          email.from.address: "{{parsed_event.message.suser}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:277]:
```suggestion
          email.from.address: ["{{parsed_event.message.suser}}"]
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,573 @@
...
+          email.to.address: "{{parsed_event.message.duser}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:278]:
```suggestion
          email.to.address: ["{{parsed_event.message.duser}}"]
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 4]
```diff
@@ -0,0 +1,573 @@
...
+          email.subject: "{{parsed_event.message.msg}}"
+          url.original: "{{parsed_event.message.request}}"
+          file.path: "{{parsed_event.message.filePath}}"
+          file.name: "{{parsed_event.message.fname}}"
+          file.size: "{{parsed_event.message.fsize}}"
+          rule.name: "{{parsed_event.message.cs5Label}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:284]:
```suggestion
          rule.name: "{{parsed_event.message.Rule}}"
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 5]
```diff
@@ -0,0 +1,573 @@
...
+
+      - translate:
+        dictionary:
+          0: USB storage device
+          1: Non-storage USB
+          2: CD/DVD
+          3: Floppy disks
+          4: Network driver
+        mapping:
+          parsed_event.message.cn2: Trend_Micro.DeviceType
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:351]:
```suggestion
          parsed_event.message.cn2: trendmicro.apexone.DeviceType
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 6]
```diff
@@ -0,0 +1,573 @@
...
+
+      - translate:
+        dictionary:
+          0: Modify
+          1: Read and execute
+          2: Read
+          3: List device content only
+          4: Block
+        mapping:
+          parsed_event.message.cn3: Trend_Micro.Permission
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:361]:
```suggestion
          parsed_event.message.cn3: trendmicro.apexone.Permission
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 7]
```diff
@@ -0,0 +1,573 @@
...
+
+  set_endpoint_application_control_event:
+    actions:
+      - set:
+          event.kind: event
+          event.type: ["change"]
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:367]:
```suggestion
          event.type: ["info"]
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 8]
```diff
@@ -0,0 +1,573 @@
...
+
+  set_engine_update_event:
+    actions:
+      - set:
+          event.kind: event
+          event.category: ["process"]
+          event.type: ["change"]
+          host.name: "{{parsed_event.message.shost}}"
+          source.ip: "{{parsed_event.message.cs2}}"
+          Trend_Micro.Engine: "{{parsed_event.message.cn2}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:397]:
```suggestion
          trendmicro.apexone.Engine: "{{parsed_event.message.cn2}}"
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 9]
```diff
@@ -0,0 +1,573 @@
...
+
+      - translate:
+        dictionary:
+          0: Unused
+          1: In use
+        mapping:
+          parsed_event.message.cn3: Trend_Micro.Engine_Status
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:404]:
```suggestion
          parsed_event.message.cn3: trendmicro.apexone.Engine_Status
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 10]
```diff
@@ -0,0 +1,573 @@
...
+          event.kind: event
+          event.category: ["network"]
+          event.type: ["connection"]
+          event.code: "{{parsed_event.message.cat}}"
+          process.executable: "{{parsed_event.message.deviceProcessName}}"
+          source.ip: "{{parsed_event.message.src}}"
+          source.port: "{{parsed_event.message.spt}}"
+          destination.ip: "{{parsed_event.message.dst}}"
+          destination.port: "{{parsed_event.message.dpt}}"
+          Trend_Micro.NCIE_ThreatName: "{{parsed_event.message.NCIE_ThreatName}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:428]:
```suggestion
          trendmicro.apexone.NCIE_ThreatName: "{{parsed_event.message.NCIE_ThreatName}}"
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 11]
```diff
@@ -0,0 +1,573 @@
...
+          event.action: "{{parsed_event.message.act}}"
+          network.direction: >-
+            {% if parsed_event.message.deviceDirection != 'None' %}
+              {{parsed_event.message.deviceDirection}}
+            {% endif %}
+
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:434]:
```suggestion
      - translate:
        dictionary:
          Pass: ["allowed"]
          Block: ["denied"]
          4: ["denied"]
        mapping:
          parsed_event.message.act: event.type
        fallback: ["access"]
         
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 12]
```diff
@@ -0,0 +1,573 @@
...
+          process.title: "{{parsed_event.message.deviceFacility}}"
+          host.name: "{{parsed_event.message.shost}}"
+          event.reason: "{{parsed_event.message.msg}}"
+
+  set_network_content_inspection_event:
+    actions:
+      - set:
+          event.kind: event
+          event.category: ["network"]
+          event.type: ["connection"]
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:421]: I suggest using `act` to compute `event.type`
```suggestion
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 13]
```diff
@@ -0,0 +1,573 @@
...
+    actions:
+      - set:
+          event.kind: alert
+          event.category: ["malware"]
+          event.type: ["info"]
+          file.name: "{{parsed_event.message.fname}}"
+          file.path: "{{parsed_event.message.filePath}}"
+          file.hash.sha1: "{{parsed_event.message.filehash}}"
+          host.name: "{{parsed_event.message.dhost}}"
+          process.user.name: "{{parsed_event.message.duser}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:472]: My bad. `process.user.name` is not defined in the ECS specification. However, `user.target.name` exists and seems more suitable
```suggestion
          user.target.name: "{{parsed_event.message.duser}}"
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 14]
```diff
@@ -0,0 +1,573 @@
...
+          host.ip: "{{parsed_event.message.dst or c6a3}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:473]: This field should be an array.

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 15]
```diff
@@ -0,0 +1,573 @@
...
+          host.name: "{{parsed_event.message.dhost}}"
+          source.ip: "{{parsed_event.message.src or parsed_event.message.c6a3}}"
+          process.name: "{{parsed_event.message.sproc}}"
+          file.hash.sha1: "{{parsed_event.message.fileHash}}"
+          file.name: "{{parsed_event.message.fname}}"
+          url.original: "{{parsed_event.message.request}}"
+          Trend_Micro.Security_Threat: "{{parsed_event.message.cs1}}"
+          Trend_Micro.Risk_Level: "{{parsed_event.message.cn1}}"
+          Trend_Micro.Threat_Categories: "{{parsed_event.message.cs2}}"
+          Trend_Micro.Cloud_Service_Vendor: "{{parsed_event.message.cs3}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:495]:
```suggestion
          trendmicro.apexone.Security_Threat: "{{parsed_event.message.cs1}}"
          trendmicro.apexone.Risk_Level: "{{parsed_event.message.cn1}}"
          trendmicro.apexone.Threat_Categories: "{{parsed_event.message.cs2}}"
          trendmicro.apexone.Cloud_Service_Vendor: "{{parsed_event.message.cs3}}"
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 16]
```diff
@@ -0,0 +1,573 @@
...
+
+  set_spyware_grayware_event:
+    actions:
+      - set:
+          event.kind: alert
+          event.category: ["malware"]
+          event.type: ["info"]
+          Trend_Micro.VirusName: "{{parsed_event.message.cs1}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:503]:
```suggestion
          trendmicro.apexone.VirusName: "{{parsed_event.message.cs1}}"
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 17]
```diff
@@ -0,0 +1,573 @@
...
+    actions:
+      - set:
+          event.kind: alert
+          event.category: ["malware"]
+          event.type: ["info"]
+          host.name: "{{parsed_event.message.dhost}}"
+          source.ip: "{{parsed_event.message.src or parsed_event.message.c6a3}}"
+          process.name: "{{parsed_event.message.sproc}}"
+          file.hash.sha1: "{{parsed_event.message.fileHash}}"
+          file.name: "{{parsed_event.message.fname}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:490]: According to the test, I suggest:
```suggestion
          file.path: "{{parsed_event.message.fname}}"
          file.name: "{{parsed_event.message.fname | basename}}"
          file.directory: "{{parsed_event.message.fname | dirname}}"
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 18]
```diff
@@ -0,0 +1,573 @@
...
+          Trend_Micro.Cloud_Service_Vendor: "{{parsed_event.message.cs3}}"
+
+  set_spyware_grayware_event:
+    actions:
+      - set:
+          event.kind: alert
+          event.category: ["malware"]
+          event.type: ["info"]
+          Trend_Micro.VirusName: "{{parsed_event.message.cs1}}"
+          file.name: "{{parsed_event.message.fname}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:504]:
```suggestion
          file.path: "{{parsed_event.message.fname}}"
          file.name: "{{parsed_event.message.fname | basename}}"
          file.directory: "{{parsed_event.message.fname | dirname}}"
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 19]
```diff
@@ -0,0 +1,573 @@
...
+          source.ip: "{{parsed_event.message.dst or parsed_event.message.c6a3}}"
+          user.name: "{{parsed_event.message.duser}}"
+
+  set_suspicious_files_event:
+    actions:
+      - set:
+          event.kind: alert
+          event.category: ["malware"]
+          event.type: ["info"]
+          file.name: "{{parsed_event.message.cs3}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:516]:
```suggestion
          file.path: "{{parsed_event.message.cs3}}"
          file.name: "{{parsed_event.message.cs3 | basename}}"
          file.directory: "{{parsed_event.message.cs3 | dirname}}"
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 20]
```diff
@@ -0,0 +1,573 @@
...
+          file.path: "{{parsed_event.message.filePath}}"
+          file.hash.sha1: "{{parsed_event.message.fileHash}}"
+          source.ip: "{{parsed_event.message.dst or parsed_event.message.c6a3}}"
+
+  set_web_security_event:
+    actions:
+      - set:
+          event.kind: event
+          event.category: ["network"]
+          event.type: ["info"]
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:541]:
```suggestion
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 21]
```diff
@@ -0,0 +1,573 @@
...
+          2: Block
+          3: Monitor
+          4: Delete
+          5: Quarantine
+          6: Warn
+          7: Warn and continue
+          8: Override
+        mapping:
+          parsed_event.message.act: event.action
+
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:564]: I suggest defining `event.type` according to the action
```suggestion

      - translate:
        dictionary:
          1: ["allowed"]
          2: ["denied"]
          4: ["denied"]
        mapping:
          parsed_event.message.act: event.type
        fallback: ["info"]
          
```

**Review by squioc — CHANGES_REQUESTED** [review]
Thanks for the job.

Please find the comment to improve the format.

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml (hunk 14 above)
Comment (naegelnajim) [reply Trend Micro/trend-micro-apex-one/ingest/parser.yml:473]: which field should be an array please ? all of them ?

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 22]
```diff
@@ -0,0 +1,294 @@
...
+      {
+        "field": "source.ip"
+      },
+      {
+        "field": "process.executable"
+      },
+      {
+        "field": "cef.event_type"
+      }
+    ]
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:40]: I suggest the following improvement
```suggestion
    "value": "Behavior monitoring: The {rule.name} raised on {host.name}",
    "conditions": [
      {
        "field": "event.dataset",
        "value": "Behavior Monitoring"
      },
      {
        "field": "rule.name"
      },
      {
        "field": "host.name"
      }
    ]
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 23]
```diff
@@ -0,0 +1,618 @@
...
+          rule.ruleset: "{{parsed_event.message.cat}}"
+
+  set_behavior_monitoring_event:
+    actions:
+      - set:
+          event.kind: event
+          event.category: ["process"]
+          host.name: "{{parsed_event.message.dvchost}}"
+          source.ip: "{{parsed_event.message.src}}"
+          process.executable: "{{parsed_event.message.sproc}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:120]: I suggest adding a custom field (must be declared in the taxonomy) to extract the target of the behavior:
```suggestion
          process.executable: "{{parsed_event.message.sproc}}"
          trendmicro.apexone.behavior.target: "{{parsed_event.message.Target}}"
```

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 24]
```diff
@@ -0,0 +1,294 @@
...
+      },
+      {
+        "field": "host.name"
+      },
+      {
+        "field": "source.ip"
+      },
+      {
+        "field": "destination.ip"
+      }
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:65]:
```suggestion
    "value": "CnC callback detected on {host.name} from {source.ip} to {destination.ip}",
    "conditions": [
      {
        "field": "event.dataset",
        "value": "CnC Callback"
      },
      {
        "field": "host.name"
      },
      {
        "field": "source.ip"
      },
      {
        "field": "destination.ip"
      }
```

File: Trend Micro/trend-micro-apex-one/ingest/parser.yml [hunk 25]
```diff
@@ -0,0 +1,618 @@
...
+          process.working_directory: "{{parsed_event.message.deviceProcessName | dirname}}"
+
+  set_content_security_event:
+    actions:
+      - set:
+          event.kind: event
+          event.category: ["email"]
+          event.type: ["info"]
+          event.severity: "{{parsed_event.message.CLF_SeverityCode}}"
+          event.code: "{{parsed_event.message.cs5Label}}"
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/ingest/parser.yml:224]:
```suggestion
          event.code: "{{parsed_event.message.cs5}}"
```

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 26]
```diff
@@ -0,0 +1,294 @@
...
+    ]
+  },
+  {
+    "value": "Content security event action taken: {event.action}.",
+    "conditions": [
+      {
+        "field": "event.action"
+      }
+    ]
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:74]:
```suggestion
    "value": "Email from {email.from.address} to {email.to.address} was filtered:{trendmicro.apexone.SL_MessageAction} ",
    "conditions": [
      {
        "field": "trendmicro.apexone.SL_MessageAction",
      },
      {
        "field": "email.from.address"
      },
      {
        "field": "email.to.address"
      }
    ]
```

**Review by squioc — COMMENTED** [review]
First review of smart-descriptions.

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 27]
```diff
@@ -0,0 +1,294 @@
...
+      },
+      {
+        "field": "process.executable"
+      },
+      {
+        "field": "file.name"
+      },
+      {
+        "field": "Trend_Micro.Permission"
+      }
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:119]:
```suggestion
    "value": "Device access: {process.name} try to access {file.name} on {Trend_Micro.DeviceType}. Permission applied: {Trend_Micro.Permission}.",
    "conditions": [
      {
        "field": "event.dataset",
        "value": "Device Access Control"
      },
      {
        "field": "Trend_Micro.DeviceType"
      },
      {
        "field": "process.executable"
      },
      {
        "field": "file.name"
      },
      {
        "field": "Trend_Micro.Permission"
      }
```

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 28]
```diff
@@ -0,0 +1,294 @@
...
+      {
+        "field": "host.name"
+      },
+      {
+        "field": "process.title"
+      },
+      {
+        "field": "event.reason"
+      }
+    ]
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:165]:
```suggestion
    "value": "Authentication on {host.name}: {event.reason}.",
    "conditions": [
      {
        "field": "host.name"
      },
      {
        "field": "event.category",
        "value": "authentication"
      },
      {
        "field": "event.reason"
      }
    ]
```

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 29]
```diff
@@ -0,0 +1,294 @@
...
+      },
+      {
+        "field": "process.executable"
+      },
+      {
+        "field": "Trend_Micro.NCIE_ThreatName"
+      },
+      {
+        "field": "event.action"
+      }
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:184]:
```suggestion
    "value": "Suspicious connection detected from {source.ip} to {destination.ip}. Threat detected: {trendmicro.apexone.NCIE_ThreatName}. Action: {event.action}",
    "conditions": [
      {
        "field": "source.ip"
      },
      {
        "field": "destination.ip"
      },
      {
        "field": "event.dataset",
        "value": "Suspicious Connection"
      },
      {
        "field": "Trend_Micro.NCIE_ThreatName"
      },
      {
        "field": "event.action"
      }
```

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 30]
```diff
@@ -0,0 +1,294 @@
...
+  {
+    "value": "Configuration change on {host.name} ({host.ip}).",
+    "conditions": [
+      {
+        "field": "host.name"
+      },
+      {
+        "field": "host.ip"
+      }
+    ]
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:196]:
```suggestion
    "value": "Configuration change on {host.name}.",
    "conditions": [
      {
        "field": "event.category",
        "value": "configuration"
      },
      {
        "field": "event.type",
        "value": "change"
      },
      {
        "field": "host.name"
      }
    ]
```

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 31]
```diff
@@ -0,0 +1,294 @@
...
+  },
+  {
+    "value": "Predictive ML Alert on {host.ip}. Executed by {user.name}.",
+    "conditions": [
+      {
+        "field": "host.ip"
+      },
+      {
+        "field": "user.name"
+      }
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:206]:
```suggestion
    "value": "Malware {file.name} detected on {host.ip}",
    "conditions": [
      {
        "field": "event.category",
        "value": "malware"
      },
      {
        "field": "host.ip"
      },
      {
        "field": "file.name"
      }
```

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 32]
```diff
@@ -0,0 +1,294 @@
...
+      },
+      {
+        "field": "Trend_Micro.VirusName"
+      },
+      {
+        "field": "file.name"
+      },
+      {
+        "field": "source.ip"
+      }
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:234]:
```suggestion
    "value": "Malware {trendmicro.apexone.VirusName} detected on {host.ip}",
    "conditions": [
      {
        "field": "event.category",
        "value": "malware"
      },
      {
        "field": "host.ip"
      },
      {
        "field": "trendmicro.apexone.VirusName"
      }
```

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 33]
```diff
@@ -0,0 +1,294 @@
...
+  {
+    "value": "Sandbox Alert on {host.name}, threat originates from {url.original}.",
+    "conditions": [
+      {
+        "field": "host.name"
+      },
+      {
+        "field": "url.original"
+      }
+    ]
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:218]:
```suggestion
    "value": "Threat {trendmicro.apexone.Security_Threat} detected on {host.ip}",
    "conditions": [
      {
        "field": "event.category",
        "value": "malware"
      },
      {
        "field": "host.ip"
      },
      {
        "field": "trendmicro.apexone.Security_Threat"
      }
```

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 34]
```diff
@@ -0,0 +1,294 @@
...
+      },
+      {
+        "field": "file.hash.sha1"
+      },
+      {
+        "field": "source.ip"
+      },
+      {
+        "field": "event.action"
+      }
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:254]:
```suggestion
    "value": "Suspicious file {file.name} detected on {host.name}. Action taken: {event.action}",
    "conditions": [
      {
        "field": "host.name"
      },
      {
        "field": "file.name"
      },
      {
        "field": "event.category",
        "value": "malware"
      },
      {
        "field": "event.dataset",
        "value": "Suspicious Files"
      },
      {
        "field": "event.action"
      }
```

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 35]
```diff
@@ -0,0 +1,294 @@
...
+        "field": "host.name"
+      },
+      {
+        "field": "user.name"
+      },
+      {
+        "field": "source.ip"
+      }
+    ]
+  },
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:270]: I think that the smart-description L199 (ie for Predictive Machine learning) should work for virus and malware events (must be checked).
```suggestion
```

File: Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json [hunk 36]
```diff
@@ -0,0 +1,294 @@
...
+      },
+      {
+        "field": "source.ip"
+      },
+      {
+        "field": "destination.ip"
+      },
+      {
+        "field": "destination.port"
+      }
```
Comment (squioc) [inline Trend Micro/trend-micro-apex-one/_meta/smart-descriptions.json:291]:
```suggestion
    "value": "Web Security Event on {url.original} from {host.name}. Action: {event.action}.",
    "conditions": [
      {
        "field": "host.name"
      },
      {
        "field": "user.name"
      },
      {
        "field": "event.action"
      },
      {
        "field": "url.original"
      }
```

**Review by squioc — APPROVED** [review]

**Pull request merged**